libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TranscriptLoggerMiddleware.java

```diff
@@ -123,6 +128,37 @@ public TranscriptLoggerMiddleware(TranscriptLogger withTranscriptLogger) {
             );
     }
 
+    private static Activity cloneActivity(Activity activity) {
+        ObjectMapper objectMapper = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .findAndRegisterModules();
+
+        try {
+            activity = objectMapper.readValue(objectMapper.writeValueAsString(activity), Activity.class);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+
+        Activity activityWithId = ensureActivityHasId(activity);
+
+        return activityWithId;
+    }
+
+    private static Activity ensureActivityHasId(Activity activity) {
+        Activity activityWithId = activity;
+
+        if (activity == null) {
+            throw new IllegalArgumentException("Cannot check or add Id on a null Activity.");
+        }
+
+        if (StringUtils.isBlank(activity.getId())) {
```

Comment:
```suggestion
        if (activity.getId() == null) {
```

C# just checks for `null`